### PR TITLE
#1179 Deprecate doctrine:query:sql command

### LINKED ISSUE #1179: Console commands: compatiblity with upcoming doctrine/dbal 3.0
Due to some deprecations and the new `ConnectionProvider` interface we need to fix a few things to be compatible with `doctrine/dbal` 3.0.

See https://github.com/doctrine/dbal/pull/3956, https://github.com/doctrine/dbal/pull/4059

Also related was this fix: https://github.com/doctrine/DoctrineBundle/pull/1170

Some thoughts: 
- we will need an adapter from `Symfony\Bridge\Doctrine\ManagerRegistry` to `Doctrine\DBAL\Tools\Console\ConnectionProvider` and inject that into the 2 commands `RunSqlDoctrineCommand` and `ImportDoctrineCommand` (can we use the dbal commands directly and deprecate the proxy commands?)
- remove usages of `Doctrine\DBAL\Tools\Console\Helper\ConnectionHelper`

### PR DESCRIPTION
**Pull request by ostrolucky**

### CODE REVIEW

File: DependencyInjection/DoctrineExtension.php [hunk 1]
```diff
@@ -197,12 +199,20 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
...
 
         // Create a shard_manager for this connection
-        if (! isset($options['shards'])) {
+        if (isset($options['shards'])) {
+            $shardManagerDefinition = new Definition($options['shardManagerClass'], [new Reference(sprintf('doctrine.dbal.%s_connection', $name))]);
+            $container->setDefinition(sprintf('doctrine.dbal.%s_shard_manager', $name), $shardManagerDefinition);
+        }
+
+        // dbal < 2.11 BC layer
+        if (! class_exists(ConnectionProvider::class)) {
```
Comment (dmaicher) [inline DependencyInjection/DoctrineExtension.php:208]: `interface_exists`

File: DependencyInjection/DoctrineExtension.php [hunk 2]
```diff
@@ -197,12 +199,20 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
...
             return;
         }
 
-        $shardManagerDefinition = new Definition($options['shardManagerClass'], [new Reference(sprintf('doctrine.dbal.%s_connection', $name))]);
-        $container->setDefinition(sprintf('doctrine.dbal.%s_shard_manager', $name), $shardManagerDefinition);
+        $container->setDefinition(
+            ManagerRegistryAwareConnectionProvider::class,
+            new Definition(ManagerRegistryAwareConnectionProvider::class, $container->getDefinition('doctrine'))
```
Comment (dmaicher) [inline DependencyInjection/DoctrineExtension.php:214]: leads to

``` 
Argument 2 passed to Symfony\Component\DependencyInjection\Definition::__construct() must be of the type array, object given

**Review by dmaicher — COMMENTED** [review]
Nice idea to deprecate the `doctrine:query:sql` command completely and not just the old proxy class :blush: 

Tested it on one of my projects and found a few issues

File: Dbal/ManagerRegistryAwareConnectionProvider.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Dbal;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Tools\Console\ConnectionProvider;
+use Doctrine\Persistence\AbstractManagerRegistry;
+
+class ManagerRegistryAwareConnectionProvider implements ConnectionProvider
```
[the remark this answers is not in the document]
Comment (ostrolucky) [reply Dbal/ManagerRegistryAwareConnectionProvider.php:9]: I don't care about that. If somebody wants to extend, I don't care

**Pull request merged**